test_src/sljitTest.c

```diff
@@ -76,17 +74,19 @@ static void *test64_ptr_hack;
 		return; \
 	}
 
-/* only relevant for test64
-   requires an sljitLir built with the right -DSLJIT_MALLOC_EXEC flag */
-SLJIT_API_FUNC_ATTRIBUTE void *sljit_test_malloc_exec(sljit_uw size)
+/* For interface testing and for test64. */
+void *sljit_test_malloc_exec(sljit_uw size, void *exec_allocator_data)
 {
-	if (SLJIT_LIKELY(!test64_ptr_hack))
-		return sljit_malloc_exec(size);
+	if (exec_allocator_data)
+		return exec_allocator_data;
 
-	if ((sljit_uw)test64_ptr_hack == 64)
-		test64_ptr_hack = sljit_malloc_exec(size);
+	return SLJIT_BUILTIN_MALLOC_EXEC(size, exec_allocator_data);
+}
 
-	return test64_ptr_hack;
+/* For interface testing. */
+void sljit_test_free_code(void* code, void *exec_allocator_data)
+{
+	SLJIT_BUILTIN_FREE_EXEC(code, exec_allocator_data);
 }
```

Comment:
should move below the !SLJIT_CONFIG_UNSUPPORTED below (won't fix the build though, but it will help)